homeassistant/components/shelly/__init__.py

```diff
@@ -296,6 +314,22 @@ def _async_device_online(_: Any, update_type: RpcUpdateType) -> None:
         LOGGER.debug("Setting up online RPC device %s", entry.title)
         try:
             await device.initialize()
+        except FirmwareUnsupported as err:
+            error = repr(err)
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                FIRMWARE_UNSUPPORTED_ISSUE_ID.format(unique=entry.unique_id),
+                is_fixable=False,
+                is_persistent=False,
+                severity=ir.IssueSeverity.ERROR,
+                translation_key="unsupported_firmware",
+                translation_placeholders={
+                    "device_name": entry.title,
+                    "ip_address": entry.data["host"],
+                },
+            )
+            raise ConfigEntryNotReady(error) from err
```

Comment:
Looks like this code is the same as above so it could probably be a function to make it more DRY

Reply:
Yeah, pushed before optimizing :-(